### PR TITLE
Add peers persistence

### PR DESCRIPTION
**Pull request by adambabik**
This PR adds a persistency for peers using sqlite.

Relates to #63.

**Do not merge. #68 contains this changes among others and only #68 should be merged.**

_I plan to create multiple PRs for #63._

### CODE REVIEW

**Review by decanus — CHANGES_REQUESTED** [review]
@adambabik do we want things like go migrate in the makefile? I believe console client had it

**Comment by adambabik** [comment]
@decanus yeah, it generates a Go file with embedded SQL migrations specific to mvds from `.sql` files. Without migrations being in a Go file, it's harder to share them and run tests as you need to deal with loading plain text files instead of using Go functions.

status-protocol-go or the console client will apply migrations defined here by importing a mvds package.

**Comment by decanus** [comment]
@adambabik can you add it to the makefile?

File: Makefile [hunk 1]
```diff
@@ -28,3 +28,7 @@ vendor:
 	go mod vendor
 	modvendor -copy="**/*.c **/*.h" -v
 .PHONY: vendor
+
+generate:
```
Comment (adambabik) [inline Makefile:32]: @decanus it's added here.

File: persistenceutil/sqlite.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/pkg/errors"
+	"github.com/status-im/migrate/v4"
+	"github.com/status-im/migrate/v4/database/sqlcipher"
+	bindata "github.com/status-im/migrate/v4/source/go_bindata"
+)
+
+// The default number of kdf iterations in sqlcipher (from version 3.0.0)
+// https://github.com/sqlcipher/sqlcipher/blob/fda4c68bb474da7e955be07a2b807bda1bb19bd2/CHANGELOG.md#300---2013-11-05
+// https://www.zetetic.net/sqlcipher/sqlcipher-api/#kdf_iter
+const defaultKdfIterationsNumber = 64000 // nolint: deadcode,varcheck,unused
```
Comment (cammellos) [inline persistenceutil/sqlite.go]: Not used?

**Review by cammellos — APPROVED** [review]

File: peers/migrations/sqlite/1565249278_initial_schema.up.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE mvds_peers (
```
Comment (cammellos) [inline peers/migrations/sqlite/1565249278_initial_schema.up.sql:1]: Could you also create an index for the usage? (`group_id`, `peer_id`)

Comment (adambabik) [reply peers/migrations/sqlite/1565249278_initial_schema.up.sql:1]: Actually there is a primary key created from these two columns. Do we need another index?

Comment (decanus) [reply peers/migrations/sqlite/1565249278_initial_schema.up.sql:1]: Might make sense to do a UNIQUE(group_id, peer_id)

Comment (cammellos) [reply peers/migrations/sqlite/1565249278_initial_schema.up.sql:1]: that's covered by the primary key, regarding the primary key and index usage https://github.com/vacp2p/mvds/pull/65#discussion_r312853747 , I think in this case it's all covered:
`PRIMARY KEY  (group_id, peer_id)`
should cover:

`WHERE group_id = ? and peer_id = ?`
`WHERE peer_id = ? and group_id = ?`
`WHERE group_id = ?`

but not
`WHERE peer_id = ?`

which I don't believe we are using, so it should be enough

Comment (adambabik) [reply peers/migrations/sqlite/1565249278_initial_schema.up.sql:1]: This is also fine, we have `PRIMARY KEY (group_id, peer_id)` and the usage is:
* `WHERE group_id = ? AND peer_id = ?`
* `WHERE group_id = ?`

**Comment by adambabik** [comment]
@decanus can you please take a look again?

**Comment by adambabik** [comment]
Closed by #68

**Pull request closed without merging**